server/src/main/java/org/elasticsearch/cluster/metadata/DataLifecycle.java

```diff
@@ -25,19 +27,52 @@
 import org.elasticsearch.xcontent.XContentParser;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Objects;
 
 /**
  * Holds the Data Lifecycle Management metadata that are configuring how a data stream is managed.
  */
 public class DataLifecycle implements SimpleDiffable<DataLifecycle>, ToXContentObject {
 
+    public static final Setting<RolloverConditions> CLUSTER_DLM_DEFAULT_ROLLOVER_SETTING = new Setting<>(
+        "cluster.dlm.default.rollover",
+        "max_age=7d,max_primary_shard_size=50gb,min_docs=1,max_primary_shard_docs=200000000",
+        (s) -> RolloverConditions.parseSetting(s, "cluster.dlm.default.rollover"),
+        new RolloverConditionsValidator(),
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
+    /**
+     * We require the default rollover conditions to have min_docs set to a non-negative number to avoid empty indices
+     * and to have at least one MAX condition set to ensure that the rollover will be triggered.
+     */
+    static class RolloverConditionsValidator implements Setting.Validator<RolloverConditions> {
```

Comment:
I'm in two minds about this validator. 
The setting is meant to allow us to make changes, it'll not be something the users will configure. 

I'm tempted to say we shouldn't add this validation, but only validate that the setting is not empty (which should be covered by the `RolloverConditions.parseSetting` ? ) 
What do you think? 


Note: If we do keep it I think it shouldn't be restricted to `min_docs`. The are other `min_*` conditions that could be valid to avoid rolling over empty indices

Reply:
> The setting is meant to allow us to make changes, it'll not be something the users will configure.
That is true, but I think even we can make a mistake, so avoiding creating empty indices is a safe bet. I see this validation helping the us not blocking us.

> If we do keep it I think it shouldn't be restricted to min_docs. The are other min_* conditions that could be valid to avoid rolling over empty indices
That is true, I did consider it but I had some concerns about the following:
- `min_age` doesn't protect us from empty indices, you can have an index that has been there for 7 days and it can still be empty.
- `min_size` & `min_primary_shard_size` concerns me because an index doesn't have an empty footprint so if you set it up wrongly you might still have an empty index (if I am not mistaken).
- `min_primary_shard_docs` I think is a good addition :).

I would suggest that we keep the validation but we add the `min_primary_shard_docs` as an option too. What do you think? I am considering that there are no use cases that would not want at least one of the two.

Reply:
Discussed offline. We will remove the checks for `min_*` conditions.

Reply:
I think the whole validator could be removed and leave the RolloverConditions#parser to make sure there's at least one condition, regardless what that is.